src/utils/replaceVariables.js

```diff
@@ -1,12 +1,46 @@
 export const variableRegEx = new RegExp(/#{(\S+)}/g);
+export const shortCodeRegEx = new RegExp(
+  /#\{(!?\S*)\s+(.*?)\}(.*)#\{\/(\S*)\}/g
+);
 const excludedValues = [">>"];
+
+const hasShortCodes = (content = "") => {
+  return content.match(shortCodeRegEx);
+};
+
+export const replaceShortCodes = (content = "", replacers = []) => {
+  let shortCodesReplaced = content;
+
+  while (hasShortCodes(shortCodesReplaced)) {
+    shortCodesReplaced = content.replace(
+      shortCodeRegEx,
+      (match, openTag, tagArgs, tagContent, closeTag, offset) => {
+        let value = content;
+        replacers.forEach((replacer) => {
+          value =
+            replacer({
+              openTag,
+              tagArgs,
+              tagContent,
+              closeTag,
```

Comment:
closeTag can be used for some sanity checks? Otherwise its just here. regex can't really check that open and close match. And if they don't, I guess we can throw an error since that's probably an authoring error?

Reply:
> regex can't really check that open and close match

I am a liar.

Reply:
That said they will match so the closeTag isn't terribly important? We'll leave it in just in case.